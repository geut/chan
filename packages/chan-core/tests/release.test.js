import toVFile from 'to-vfile'
import { advanceTo } from 'jest-date-mock'
import { dirname } from 'dirname-filename-esm'

import { addRelease } from '../src/index.js'

beforeEach(() => {
  advanceTo(new Date(2019, 0, 11))
})

test('add first release with url', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/unreleased.md`), {
    version: '0.0.1',
    gitReleaseTemplate: 'https://github.com/geut/chan/releases/tag/[next]',
    gitCompareTemplate: 'https://github.com/geut/chan/compare/[prev]...[next]',
    gitBranch: 'HEAD'
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add first release without url', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/unreleased.md`), {
    version: '0.0.1'
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add release with url', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/used.md`), {
    version: '0.0.5',
    gitCompareTemplate: 'https://github.com/geut/chan/compare/[prev]...[next]',
    gitBranch: 'HEAD'
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add release without url', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/used.md`), {
    version: '0.0.5'
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add release yanked', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/used.md`), {
    version: '0.0.5',
    gitCompareTemplate: 'https://github.com/geut/chan/compare/[prev]...[next]',
    gitBranch: 'HEAD',
    yanked: true
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add release allowing prereleases', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/prereleases.md`), {
    version: '0.0.2-beta.3',
    gitCompareTemplate: 'https://github.com/geut/chan/compare/[prev]...[next]',
    gitBranch: 'HEAD',
    allowPrerelease: true
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add release merged with prereleases', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/prereleases.md`), {
    version: '0.0.2',
    gitCompareTemplate: 'https://github.com/geut/chan/compare/[prev]...[next]',
    gitBranch: 'HEAD',
    mergePrerelease: true
  })

  expect(file.toString()).toMatchSnapshot()
})

test('add release with altered prefix', async () => {
  const file = await addRelease(toVFile.readSync(`${dirname(import.meta)}/__files__/unreleased.md`), {
    version: '0.0.1',
    gitReleaseTemplate: 'https://github.com/geut/chan/releases/tag/[next]',
    gitCompareTemplate: 'https://github.com/geut/chan/compare/[prev]...[next]',
    gitBranch: 'HEAD',
    releasePrefix: ''
  })

  expect(file.toString()).toMatchSnapshot()
})
