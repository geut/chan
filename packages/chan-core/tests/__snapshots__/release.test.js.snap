// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add first release with url 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.0.1] - 2019-01-11
### Changed
- Changed X by Y.

[Unreleased]: https://github.com/geut/chan/compare/v0.0.1...HEAD
[0.0.1]: https://github.com/geut/chan/releases/tag/v0.0.1
"
`;

exports[`add first release without url 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## 0.0.1 - 2019-01-11
### Changed
- Changed X by Y.
"
`;

exports[`add release allowing prereleases 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.0.2-beta.3] - 2019-01-11
### Changed
- Something on 0.0.2

## [0.0.2-beta.2] - 2019-01-18
### Added
- feature from 0.0.2-beta.2

### Security
- feature from 0.0.2-beta.2

### Fixed
- feature from 0.0.2-beta.2

## [0.0.2-beta.1] - 2019-01-17
### Added
- feature from 0.0.2-beta.1

### Fixed
- feature from 0.0.2-beta.1

## 0.0.1 - 2019-01-18
### Added
- feature from 0.0.1

[Unreleased]: https://github.com/geut/chan/compare/v0.0.2-beta.3...HEAD
[0.0.2-beta.3]: https://github.com/geut/chan/compare/v0.0.2-beta.2...v0.0.2-beta.3
[0.0.2-beta.2]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.2-beta.1...v0.0.2-beta.2
[0.0.2-beta.1]: hhttps://github.com/olivierlacan/keep-a-changelog/compare/v0.0.1...v0.0.2-beta.1
"
`;

exports[`add release merged with prereleases 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.0.2] - 2019-01-11
### Changed
- Something on 0.0.2

### Added
- feature from 0.0.2-beta.2
- feature from 0.0.2-beta.1

### Security
- feature from 0.0.2-beta.2

### Fixed
- feature from 0.0.2-beta.2
- feature from 0.0.2-beta.1

## 0.0.1 - 2019-01-18
### Added
- feature from 0.0.1

[Unreleased]: https://github.com/geut/chan/compare/v0.0.2...HEAD
[0.0.2]: https://github.com/geut/chan/compare/v0.0.1...v0.0.2
"
`;

exports[`add release with altered prefix 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.0.1] - 2019-01-11
### Changed
- Changed X by Y.

[Unreleased]: https://github.com/geut/chan/compare/0.0.1...HEAD
[0.0.1]: https://github.com/geut/chan/releases/tag/0.0.1
"
`;

exports[`add release with url 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.0.5] - 2019-01-11
### Changed
- package1
  - test1
  - test2
- Update and improvement of Polish translation from [@m-aciek](https://github.com/m-aciek).

## [0.0.4] - 2014-08-09
### Added
- Better explanation of the difference between the file (\\"CHANGELOG\\")
  and its function \\"the change log\\".

### Changed
- Refer to a \\"change log\\" instead of a \\"CHANGELOG\\" throughout the site
  to differentiate between the file and the purpose of the file — the
  logging of changes.

### Removed
- Remove empty sections from CHANGELOG, they occupy too much space and
  create too much noise in the file. People will have to assume that the
  missing sections were intentionally left out because they contained no
  notable changes.

## 0.0.3 - 2014-08-09 [YANKED]

## [0.0.2] - 2014-07-10
### Added
- Explanation of the recommended reverse chronological release ordering.

## 0.0.1 - 2014-05-31
### Added
- This CHANGELOG file to hopefully serve as an evolving example of a
  standardized open source project CHANGELOG.
- CNAME file to enable GitHub Pages custom domain
- README now contains answers to common questions about CHANGELOGs
- Good examples and basic guidelines, including proper date formatting.
- Counter-examples: \\"What makes unicorns cry?\\"

[Unreleased]: https://github.com/geut/chan/compare/v0.0.5...HEAD
[0.0.5]: https://github.com/geut/chan/compare/v0.0.4...v0.0.5
[0.0.4]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.3...v0.0.4
[0.0.2]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.1...v0.0.2
"
`;

exports[`add release without url 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## 0.0.5 - 2019-01-11
### Changed
- package1
  - test1
  - test2
- Update and improvement of Polish translation from [@m-aciek](https://github.com/m-aciek).

## [0.0.4] - 2014-08-09
### Added
- Better explanation of the difference between the file (\\"CHANGELOG\\")
  and its function \\"the change log\\".

### Changed
- Refer to a \\"change log\\" instead of a \\"CHANGELOG\\" throughout the site
  to differentiate between the file and the purpose of the file — the
  logging of changes.

### Removed
- Remove empty sections from CHANGELOG, they occupy too much space and
  create too much noise in the file. People will have to assume that the
  missing sections were intentionally left out because they contained no
  notable changes.

## 0.0.3 - 2014-08-09 [YANKED]

## [0.0.2] - 2014-07-10
### Added
- Explanation of the recommended reverse chronological release ordering.

## 0.0.1 - 2014-05-31
### Added
- This CHANGELOG file to hopefully serve as an evolving example of a
  standardized open source project CHANGELOG.
- CNAME file to enable GitHub Pages custom domain
- README now contains answers to common questions about CHANGELOGs
- Good examples and basic guidelines, including proper date formatting.
- Counter-examples: \\"What makes unicorns cry?\\"

[Unreleased]: https://github.com/olivierlacan/keep-a-changelog/compare/v1.0.0...HEAD
[0.0.4]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.3...v0.0.4
[0.0.2]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.1...v0.0.2
"
`;

exports[`add release yanked 1`] = `
"# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]
### Changed
- package1
  - test1
  - test2
- Update and improvement of Polish translation from [@m-aciek](https://github.com/m-aciek).

## 0.0.5 - 2019-01-11 [YANKED]

## [0.0.4] - 2014-08-09
### Added
- Better explanation of the difference between the file (\\"CHANGELOG\\")
  and its function \\"the change log\\".

### Changed
- Refer to a \\"change log\\" instead of a \\"CHANGELOG\\" throughout the site
  to differentiate between the file and the purpose of the file — the
  logging of changes.

### Removed
- Remove empty sections from CHANGELOG, they occupy too much space and
  create too much noise in the file. People will have to assume that the
  missing sections were intentionally left out because they contained no
  notable changes.

## 0.0.3 - 2014-08-09 [YANKED]

## [0.0.2] - 2014-07-10
### Added
- Explanation of the recommended reverse chronological release ordering.

## 0.0.1 - 2014-05-31
### Added
- This CHANGELOG file to hopefully serve as an evolving example of a
  standardized open source project CHANGELOG.
- CNAME file to enable GitHub Pages custom domain
- README now contains answers to common questions about CHANGELOGs
- Good examples and basic guidelines, including proper date formatting.
- Counter-examples: \\"What makes unicorns cry?\\"

[Unreleased]: https://github.com/geut/chan/compare/v0.0.5...HEAD
[0.0.4]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.3...v0.0.4
[0.0.2]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.0.1...v0.0.2
"
`;
